### PR TITLE
Adding aud claim verification for internal api key.

### PR DESCRIPTION
**Pull request by rashm1n**
### Purpose
Adding aud claim verification for internal api key.

### Issues

Fixes
* https://github.com/wso2-enterprise/choreo/issues/20799

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyHandler.java [hunk 1]
```diff
@@ -91,6 +93,25 @@ public void checkInRevokedMap(String tokenIdentifier, String[] splitToken) throw
...
+     *
+     * @param tokenAudience list of intended audience of the API Key
+     * @param deploymentType Deployment type of the API
+     * @throws APISecurityException The comparison b/w the audience vs deployment type is invalid
+     */
+    public void compareAudience(List<String> tokenAudience, String deploymentType) throws APISecurityException {
+        if (tokenAudience != null && tokenAudience.size() > 0 &&
+                !StringUtils.isEmpty(deploymentType)) {
+            if (!tokenAudience.get(0).toLowerCase().contains(deploymentType.toLowerCase())) {
+                    throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyHandler.java:107]: It should be unauthorized

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyHandler.java [hunk 2]
```diff
@@ -91,6 +93,25 @@ public void checkInRevokedMap(String tokenIdentifier, String[] splitToken) throw
...
+     * Compares the audience claim with the deployment type of the API.
+     *
+     * @param tokenAudience list of intended audience of the API Key
+     * @param deploymentType Deployment type of the API
+     * @throws APISecurityException The comparison b/w the audience vs deployment type is invalid
+     */
+    public void compareAudience(List<String> tokenAudience, String deploymentType) throws APISecurityException {
+        if (tokenAudience != null && tokenAudience.size() > 0 &&
+                !StringUtils.isEmpty(deploymentType)) {
+            if (!tokenAudience.get(0).toLowerCase().contains(deploymentType.toLowerCase())) {
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyHandler.java:106]: Can we guarantee that the claim is sent as the 0th element, or should we iterate

Comment (rashm1n) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyHandler.java:106]: The aud claim was added as a task for this specific issue. So right now it's sent as the 0th and only element of aud array. WDYT ? @VirajSalaka

Comment (VirajSalaka) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyHandler.java:106]: ack

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/constants/APISecurityConstants.java [hunk 3]
```diff
@@ -168,5 +168,8 @@ public static String getFailureMessageDetailDescription(int errorCode, String er
 
     public static final int DEFAULT_MAX_VALID_KEYS = 250;
     public static final int DEFAULT_MAX_INVALID_KEYS = 100;
+
+    public static final int API_AUTH_INVALID_AUDIENCE = 901104;
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/constants/APISecurityConstants.java:172]: shall we reuse the 901102 status code here as per discussion

Comment (rashm1n) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/constants/APISecurityConstants.java]: Done via https://github.com/wso2/product-microgateway/pull/3357/commits/55cc1cf5b40ccd3a3d161aa6ae790780a8d79264

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230526.11

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230526.11

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230526.11

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : prod-deployment-v2 : 20230529.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230529.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230529.3

**Pull request merged**